beefy-gadget/src/lib.rs

```diff
@@ -231,13 +235,20 @@ where
 	}
 
 	fn handle_finality_notification(&mut self, notification: FinalityNotification<Block>) {
-		debug!(target: "beefy", "🥩 Finality notification: {:?}", notification);
+		debug!(target: "beefy", "Finality notification: {:?}", notification);
 
 		if self.should_vote_on(*notification.header.number()) {
-			// TODO: this needs to be properly populated by signing an MMR root as the payload
-			// (and/or abstracting the "thing to sign") and with support for validator set changes.
+			let mmr_root = if let Some(hash) = find_mmr_root_digest::<Block, Id>(&notification.header) {
+				hash
+			} else {
+				debug!(target: "beefy", "No MMR root digest found for: {:?}", notification.header.hash());
```

Comment:
Maybe make it a warning?
```suggestion
				warn!(target: "beefy", "🥩 No MMR root digest found for: {:?}", notification.header.hash());
```